app/components/close_help_post.py

```diff
@@ -19,23 +24,58 @@
 )
 
 
-async def mention_entity(entity_id: int) -> str | None:
-    output = await entity_message(
-        # Forging a message to use the entity mention logic
-        cast("dc.Message", SimpleNamespace(content=f"#{entity_id}"))
-    )
-    return output.content or None
-
+@final
+@app_commands.guild_only()
+class Close(commands.GroupCog, group_name="close"):
+    def __init__(self, bot: GhosttyBot) -> None:
+        self.description = "Mark current post as resolved."
+        self.bot = bot
+
+    @override
+    def interaction_check(self, interaction: dc.Interaction, /) -> bool:
+        user = interaction.user
+        if is_dm(user):
+            return False
+        if not (
+            isinstance((post := interaction.channel), dc.Thread)
+            and post.parent_id == self.bot.config.help_channel_id
+        ):
+            # Can only close posts in #help
+            return False
+
+        # Only helpers and the author can close posts
+        return is_mod(user) or is_helper(user) or user.id == post.owner_id
+
+    @override
+    async def cog_app_command_error(
+        self, interaction: dc.Interaction, error: app_commands.AppCommandError
+    ) -> None:
+        if type(error) is app_commands.CheckFailure:
+            # Triggers if self.interaction_check fails
+            await interaction.response.send_message(
+                "This command can only be used in "
+                f"<#{self.bot.config.help_channel_id}> "
+                "posts, by helpers or the post's author.",
+                ephemeral=True,
+            )
+            interaction.extras["error_handled"] = True
+            return
+        raise error
```

Comment:
```suggestion
        if type(error) is not app_commands.CheckFailure:
            raise error
        # Triggers if self.interaction_check fails
        await interaction.response.send_message(
            f"This command can only be used in <#{self.bot.config.help_channel_id}> "
            "posts, by helpers or the post's author.",
            ephemeral=True,
        )
        interaction.extras["error_handled"] = True
```
Also, I notice this fragment uses `bot.config.help_channel_id`, while `convert_nitro_emojis` uses `bot.ghostty_guild.id`. Let's stick to one source and either do `bot.help_channel.id` here, or change the other one to be `bot.config.guild_id`.

Reply:
We can change it for consistency. But `bot.ghostty_guild.id` is not the same thing as `bot.config.ghostty_guild_id`. This is because `ghostty_guild_id` is an optional env. So in the convert nitro section, the use of `bot.ghostty_guild.id` is the only correct semantic. 

That said, the others can be converted to match that pattern.

Reply:
True, I forgot about that part.